edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/MetricsParser.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Newtonsoft.Json;
+
+    public static class MetricsParser
+    {
+        const string PrometheusMetricSchema = @"^(?<metricname>[^#\{\}]+)(\{((?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\"")(,(?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\""))*)\})?\s(?<metricvalue>.+)$";
+        static readonly Regex PrometheusSchemaRegex = new Regex(PrometheusMetricSchema, RegexOptions.Compiled);
+
+        public static IList<Metric> ParseMetrics(DateTime timeGeneratedUtc, string prometheusMessage)
+        {
+            var metricsDataList = new List<Metric>();
+            using (StringReader sr = new StringReader(prometheusMessage))
+            {
+                string line;
+                while ((line = sr.ReadLine()) != null)
```

Comment:
What would happen here if a line of metrics was corrupted or in an invalid format? Does your code handle this and drop the line and move on?

Reply:
yea, the ```if (match.Success)``` only continues to read that line if it is valid.